File: ds/js/deque.js
class Deque {
    constructor() {
        this.deque = []
    }

    isEmpty() { 
        return this.deque.length ? false : true
    }

    enqueueFront(item) {
        this.deque.splice(0, 0, item)
    }

    enqueueRear(item) {
        this.deque.push(item)
    }

    dequeueFront() {
        return this.deque.shift()
    }

    dequeueRear() {
        return this.deque.pop()
    }
}


const deque = new Deque()

console.log(deque.isEmpty())

deque.enqueueFront(5)
deque.enqueueFront(4)
deque.enqueueRear(6)
deque.enqueueRear(7)
deque.enqueueFront(3)

console.log(deque.isEmpty())
console.log(deque)


deque.dequeueRear()
deque.dequeueFront()
deque.dequeueRear()

console.log(deque)

File: ds/python/deque.py
'''Deque using list'''

class Deque:
    def __init__(self):
        self.deque = []
    
    def is_empty(self):
        return self.deque == []
    
    def enqueue_rear(self, item):
        self.deque.append(item)

    def enqueue_front(self, item):
        self.deque.insert(0, item)
    
    def dequeue_rear(self):
        return self.deque.pop()

    def dequeue_front(self):
        return self.deque.pop(0)
    
    def size(self):
        return len(self.deque)
    

d = Deque()
print(d.is_empty())
d.enqueue_rear(8)
d.enqueue_rear(5)
d.enqueue_front(7)
d.enqueue_front(10)
print(d.size())
print(d.is_empty())
d.enqueue_rear(11)
print(d.dequeue_rear())
print(d.dequeue_front())
d.enqueue_front(55)
d.enqueue_rear(45)
print(d.deque)